tests/FaciesDescriptionControllerTest.cs

```diff
@@ -0,0 +1,206 @@
+        Assert.AreEqual(40, faciesDescription.FromDepth);
+        Assert.AreEqual(50, faciesDescription.ToDepth);
+        Assert.AreEqual("radical Technician Personal Loan Account", faciesDescription.Description);
+        Assert.AreEqual(9002, faciesDescription.QtDescriptionId);
+        Assert.AreEqual(6_000_001, faciesDescription.StratigraphyId);
+    }
+
+    [TestMethod]
+    public async Task EditFaciesDescriptionWithCompleteFaciesDescription()
+    {
+        var id = 10_000_039;
+        var originalFaciesDescription = new FaciesDescription
+        {
+            CreatedBy = null,
+            CreatedById = 4,
+            Created = new DateTime(2021, 12, 23, 8, 57, 51).ToUniversalTime(),
+            Description = "bandwidth impactful connecting",
+            FromDepth = 90,
+            ToDepth = 100,
+            Id = 10_000_039,
+            IsLast = true,
+            QtDescription = null,
+            QtDescriptionId = 9003,
+            Updated = new DateTime(2021, 11, 14, 8, 3, 19).ToUniversalTime(),
+            UpdatedBy = null,
+            UpdatedById = 2,
+            StratigraphyId = 6_000_003,
+        };
+
+        var newFaciesDescription = new FaciesDescription
+        {
+            Id = id,
+            CreatedById = 3,
+            UpdatedById = 3,
+            Created = new DateTime(2021, 3, 31, 16, 55, 02).ToUniversalTime(),
+            StratigraphyId = 6_000_010,
+            Description = "solid state web-enabled Maryland",
+            QtDescriptionId = 9001,
+        };
+
+        var faciesDescriptionToEdit = context.FaciesDescriptions.Single(c => c.Id == id);
+        Assert.AreEqual(4, faciesDescriptionToEdit.CreatedById);
+        Assert.AreEqual(2, faciesDescriptionToEdit.UpdatedById);
+        Assert.AreEqual(6_000_003, faciesDescriptionToEdit.StratigraphyId);
+        Assert.AreEqual("bandwidth impactful connecting", faciesDescriptionToEdit.Description);
+
+        // Upate FaciesDescription
+        var response = await controller.EditAsync(newFaciesDescription);
+        var okResult = response as OkObjectResult;
+        Assert.AreEqual(200, okResult.StatusCode);
+
+        // Assert Updates and unchanged values
+        var updatedContext = ContextFactory.CreateContext();
+        var updatedFaciesDescription = updatedContext.FaciesDescriptions.Single(c => c.Id == id);
+
+        Assert.AreEqual(3, updatedFaciesDescription.CreatedById);
+        Assert.AreEqual(1, updatedFaciesDescription.UpdatedById);
+        Assert.AreEqual(6_000_010, updatedFaciesDescription.StratigraphyId);
+        Assert.AreEqual("solid state web-enabled Maryland", updatedFaciesDescription.Description);
+        Assert.AreEqual(9001, updatedFaciesDescription.QtDescriptionId);
+
+        // Reset edits
+        var cleanupContext = ContextFactory.CreateContext();
+        cleanupContext.Update(originalFaciesDescription);
+        await cleanupContext.SaveChangesAsync();
```

Comment:
PP: Den Reset in ein `try-finally`-Block packen, so dass der Reset auch im Fehlerfall ausgeführt wird.